src/tarantool-gdb.py

```diff
@@ -0,0 +1,1397 @@
+        COMPRESSION_TYPE_MAX = gdb.parse_and_eval('compression_type_MAX')
+        compression_type_strs = gdb.parse_and_eval('compression_type_strs')
+
+        def __init__(self, val):
+            self.val = val
+
+        def __str__(self):
+            type = self.val['type']
+            if type >= self.COMPRESSION_TYPE_NONE and type < self.COMPRESSION_TYPE_MAX:
+                type = self.compression_type_strs[type].string()
+            return 'compression({}):[{}]->[{}]'.format(type, self.val['raw_size'], self.val['size'])
+
+if find_type('struct interval') is not None:
+    class Interval:
+        def __init__(self):
+            self.year = 0
+            self.month = 0
+            self.week = 0
+            self.day = 0
+            self.hour = 0
+            self.min = 0
+            self.sec = 0
+            self.nsec = 0
+            self.adjust = 0
+
+        def __str__(self): # interval_to_string
+            signed_fmt = [
+                '{:d}',
+                '{:+d}',
+            ]
+            s = []
+            if self.year != 0:
+                s.append(signed_fmt[len(s) == 0].format(self.year) + ' years')
+            if self.month != 0:
+                s.append(signed_fmt[len(s) == 0].format(self.month) + ' months')
+            if self.week != 0:
+                s.append(signed_fmt[len(s) == 0].format(self.week) + ' weeks')
+            if self.day != 0:
+                s.append(signed_fmt[len(s) == 0].format(self.day) + ' days')
+            if self.hour != 0:
+                s.append(signed_fmt[len(s) == 0].format(self.hour) + ' hours')
+            if self.min != 0:
+                s.append(signed_fmt[len(s) == 0].format(self.min) + ' minutes')
+            if self.sec != 0 or len(s) == 0:
+                s.append(signed_fmt[len(s) == 0].format(self.sec) + ' seconds')
+            if self.nsec != 0:
+                s.append(signed_fmt[len(s) == 0].format(self.nsec) + ' nanoseconds')
+            return ' '.join(s)
+
+    class IntervalPrinter:
+        def __init__(self, val):
+            self.val = Interval(val)
+
+        def to_string(self):
+            return str(self.val)
+
+    pp.add_printer('Interval', '^interval$', IntervalPrinter)
+
+
+class MsgPackPrint(gdb.Command):
+    '''
+Decode and print MsgPack referred by EXP in a human-readable form
+Usage: tt-mp EXP [DEPTH [MAXLENGTH]]
+    '''
+    def __init__(self):
+        super(MsgPackPrint, self).__init__('tt-mp', gdb.COMMAND_DATA)
+
+    def invoke(self, arg, from_tty):
+        argv = gdb.string_to_argv(arg)
+        argc = len(argv)
+        if argc < 1:
+            raise gdb.GdbError('MsgPack is missing')
+        mp = TtMsgPack(gdb.parse_and_eval(argv[0]))
+        if argc > 2:
+            s = mp.to_string(int(argv[1]), int(argv[2]))
+        elif argc > 1:
+            s = mp.to_string(int(argv[1]))
+        else:
+            s = mp.to_string()
+        gdb.write(s + '\n')
```

Comment:
There is a problem with expressions that contain a space.

```gdb
(gdb) tt-mp key
[1]
(gdb) tt-mp (char *)key
Traceback (most recent call last):
  File "/home/alex/tools/tmp/tarantool-gdb.py", line 1208, in invoke
    mp = TtMsgPack(gdb.parse_and_eval(argv[0]))
gdb.error: A syntax error in expression, near `'.
Error occurred in Python: A syntax error in expression, near `'.
```

Reply:
will be fixed later